numba/unicode.py

```diff
@@ -498,6 +510,14 @@ def find_impl(a, b):
         return find_impl
 
 
+@overload_method(types.UnicodeType, 'rfind')
+def unicode_rfind(src, sub, start=0, end=-1):
+    if isinstance(sub, types.UnicodeType):
```

Comment:
This also needs to check that `start` and `end` are `types.Integer`, `int`, `Optional` on type `types.Integer`, `types.NoneType`, `None` etc. Recall that `start` and `end` are interpreted in slice notation so `None` is also valid.

Reply:
Good point!I will add an additional check for the types of start and end arguments

Reply:
Thanks. It's worth keeping in mind that there is a general expectation that a feature is either complete or will gracefully fail with such limitations declared.